### PR TITLE
Escaped Javadoc special chars

### PR DESCRIPTION
**Pull request by puneetbehl**
Escaping of special chars in Javadoc, such as '<' with '&lt;' Refactoring of code -> Define explicit types and remove unnecessary variable usages.

### CODE REVIEW

File: grails-encoder/src/main/groovy/org/grails/buffer/StreamCharBuffer.java [hunk 1]
```diff
@@ -101,13 +101,11 @@
...
- * <code>
- * out << body()
- * </code><br>
+ * <code>out &lt;&lt; body()</code>
+ * <br>
  * other example:<br>
- * <code>
- * out << g.render(template: '/some/template', model:[somebean: somebean])
- * </code><br>
+ * <code>out &lt;%lt g.render(template: '/some/template', model:[somebean: somebean])</code>
```
Comment (mattmoss) [inline grails-encoder/src/main/groovy/org/grails/buffer/StreamCharBuffer.java]: Should be: `out &lt;&lt; g.render`

File: grails-web-common/src/main/groovy/org/grails/web/json/JSONObject.java [hunk 2]
```diff
@@ -796,7 +796,7 @@ public JSONObject putOpt(String key, Object value) throws JSONException {
 
     /**
      * Produce a string in double quotes with backslash sequences in all the
-     * right places. A backslash will be inserted within </, allowing JSON
+     * right places. A backslash will be inserted within $lt;/, allowing JSON
```
Comment (mattmoss) [inline grails-web-common/src/main/groovy/org/grails/web/json/JSONObject.java]: Should be: `&lt;`

File: grails-databinding/src/main/groovy/grails/databinding/SimpleDataBinder.groovy [hunk 3]
```diff
@@ -260,14 +261,14 @@ class SimpleDataBinder implements DataBinder {
     }
 
     protected boolean isOkToBind(String propName, List whiteList, List blackList) {
-        'class' != propName && 'classLoader' != propName && 'protectionDomain' != propName && 'metaClass' != propName && !blackList?.contains(propName) && (!whiteList || whiteList.contains(propName) || whiteList.find { it -> it?.toString()?.startsWith(propName + '.')})
+        'class' != propName && 'classLoader' != propName && 'protectionDomain' != propName && 'metaClass' != propName && 'metaPropertyValues' != propName && 'properties' != propName && !blackList?.contains(propName) && (!whiteList || whiteList.contains(propName) || whiteList.find { it -> it?.toString()?.startsWith(propName + '.')})
```
Comment (mattmoss) [inline grails-databinding/src/main/groovy/grails/databinding/SimpleDataBinder.groovy:264]: Do you want to include this change, and the following change in this file, as part of this commit or in a commit of its own?

**Review by mattmoss — CHANGES_REQUESTED** [review]

File: grails-databinding/src/main/groovy/grails/databinding/SimpleDataBinder.groovy (hunk 3 above)
Comment (puneetbehl) [reply grails-databinding/src/main/groovy/grails/databinding/SimpleDataBinder.groovy:264]: Yes, I would like to include it as a part of this commit.

**Pull request merged**